win32ss/gdi/gdi32/objects/font.c

```diff
@@ -304,15 +304,19 @@ IntEnumFontFamilies(HDC Dc, LPLOGFONTW LogFont, PVOID EnumProc, LPARAM lParam,
         return 1;
     }
 
+    ZeroMemory(&lfW, sizeof(lfW));
     if (!LogFont)
     {
         lfW.lfCharSet = DEFAULT_CHARSET;
-        lfW.lfPitchAndFamily = 0;
-        lfW.lfFaceName[0] = 0;
-        LogFont = &lfW;
+    }
+    else
+    {
+        lfW.lfCharSet = LogFont->lfCharSet;
+        lfW.lfPitchAndFamily = LogFont->lfPitchAndFamily;
+        StringCbCopyW(lfW.lfFaceName, LF_FACESIZE, LogFont->lfFaceName);
```

Comment:
LF_FACESIZE is the number of characters. You want `sizeof(lfW.lfFaceName)` or `StringCchCopyW`.

Reply:
OK.